docs/source/extension/extension_migration.rst

```diff
@@ -32,6 +32,19 @@ breaking change when defining fixture values that are array of objects.
 
 See the `Playwright 1.46.0 release notes <https://github.com/microsoft/playwright/releases/tag/v1.46.0>`_ for more information.
 
+
+Shared model
+~~~~~~~~~~~~
+
+The outputs set on the shared cell model are now expected to be wrapped
+in the pycrdt ``Map`` objects rather than provided as plain dictionaries.
+The use of dictionaries is deprecated and will stop working in a future version.
+For reference, see PRs:
+
+- https://github.com/jupyterlab/jupyterlab/pull/16498
+- https://github.com/jupyter-server/jupyter_ydoc/pull/241
```

Comment:
@davidbrochart is this correct? Or should we make this an implementation detail and fix this on `jupyter_ydoc` level by modifying the setter of `outputs` to do the wrapping for the API user? It may be hard because I think cells are just `YMap` for now, but I guess for sake of better typing we might want to have a subclass with custom setters - is this possible?

For reference I also opened https://github.com/jupyter-server/jupyter_ydoc/issues/269 to track typing enhancements.

Reply:
This is correct, but not complete. The reason each output is now a `Y.Map<any>` is that if the output is of type `"stream"`, its `"text"` entry is of type `Y.Array<string>`. This is also why it's difficult to type an output more precisely than `Y.Map<any>`: the type is dynamic, as it depends on the output's type which is only known at run-time. I agree that we could at least type it as a union of two types, one corresponding to a `"stream"` output type and the other to other types of outputs.